azext_iot/iothub/params.py

```diff
@@ -249,6 +249,98 @@ def load_iothub_arguments(self, _):
             help="MIME Type of file.",
         )
 
+    with self.argument_context("iot edge devices") as context:
+        context.argument(
+            "devices",
+            options_list=["--device", "-d"],
+            nargs="+",
+            action="append",
+            help="Space-separated key=value pairs corresponding to properties of the edge device to create. "
+            "The following key values are supported: `id` (device_id), `deployment` (inline json or path to file), `hostname`, "
+            "`parent` (device_id), `edge_agent` (image URL), and `container_auth` (inline json or path to file). "
+            "--device can be used 1 or more times. Review help examples for full parameter usage  - these parameters also refer "
+            "to their corresponding values in our sample configuration file: "
+            "https://github.com/Azure/azure-iot-cli-extension/tree/dev/docs/samples/sample_devices_config.yaml"
+        )
+        context.argument(
+            "clean",
+            options_list=["--clean", "-c"],
+            arg_type=get_three_state_flag(),
+            help="Deletes all devices in target hub before creating new devices.",
+        )
+        context.argument(
+            "visualize",
+            options_list=["--visualize", "--vis", "-v"],
+            arg_type=get_three_state_flag(),
+            help="Shows visualizations of devices and progress of various tasks "
+            "(device creation, setting parents, updating configs, etc).",
+        )
+        context.argument(
+            "config_file",
+            options_list=["--config-file", "--config", "--cfg"],
+            help="Path to devices configuration file. Sample configuration file: "
+            "https://github.com/Azure/azure-iot-cli-extension/tree/dev/docs/samples/sample_devices_config.yaml",
+        )
+        context.argument(
+            "device_auth_type",
+            arg_type=get_enum_type(
+                [DeviceAuthType.shared_private_key.value, DeviceAuthType.x509_thumbprint.value]
+            ),
+            options_list=["--device-auth-type", "--device-auth"],
+            help="Device to hub authorization mechanism.",
+        )
+        context.argument(
+            "default_edge_agent",
+            options_list=["--default-edge-agent", "--default-agent", "--dea"],
+            help="Default edge agent for created Edge devices if not specified individually.",
+        )
+        context.argument(
+            "device_config_template",
+            options_list=["--device-template", "--dt"],
```

Comment:
Just an observation that `--device-template`  is somewhat overloaded,  in that "device templates" are used in central [1](https://learn.microsoft.com/en-us/azure/iot-central/core/concepts-device-templates) [2](https://github.com/Azure/azure-iot-cli-extension/issues/364), DTDL (where it transitioned to `model` as the preferred term) and ...